addons/website/models/res_users.py

```diff
@@ -87,3 +87,12 @@ def authenticate(cls, db, login, password, user_agent_env):
                 visitor_pre_authenticate_sudo.access_token = user_partner.id
                 visitor_pre_authenticate_sudo._update_visitor_last_visit()
         return uid
+
+    def write(self, vals):
+        # If the user was a portal and becomes internal, remove website_id.
+        portal_users = self.filtered(lambda user: user._is_portal())
+        ret = super().write(vals)
```

Comment:
```suggestion
        res = super().write(vals)
```
Nitpicking but let's stick to the standard, it kinda "pique les yeux" :innocent: 

![image](https://github.com/odoo/odoo/assets/30048408/b1683f25-81d7-4187-b9a9-cc71adf6c7ba) ![image](https://github.com/odoo/odoo/assets/30048408/6146f788-3f0f-4b82-b8f5-dfdf06100a58)